godaddy/gasket
# Redux refresh

## Summary



Major changes tuning Redux support up for the latest Next.js + next-redux-wrapper changes.

## Changelog

*@gasket/plugin-redux*
- Generate a redux/store.js with support for adding reducers via plugins
- Set process.env.GASKET_MAKE_STORE_FILE for shared package imports

*@gasket/plugin-nextjs*
- Generate pages/_app.s with next-redux-wrapper setup from store
- Generate a redux/store.js with next-redux-wrapper setup

*@gasket/redux*
- Support for adding an entry root reducer
- No longer provide a default make-store



## Test Plan


- Additional unit tests
- Local app testing

Local testing is a bit messy. This is what I did.

To test **create**:
```
gasket create my-app -p @gasket/nextjs --plugins @gasket/redux@file:/path/to/gasket/packages/gasket-plugin-redux,@gasket/nextjs@file:/path/to/gasket/packages/gasket-plugin-nextjs
```

To test **build**/**start** and avoid duplicate react errors...
in package.json, change `file:/` version to `^5.6.0` for the above plugins and reinstall
```
rm -rf node_modules
npm i
```
Then you can:
```
cp /path/to/gasket/packages/gasket-plugin-redux/lib/* node_modules/@gasket/plugin-redux/lib
cp /path/to/gasket/packages/gasket-redux/lib/* node_modules/@gasket/redux/lib
npm run build && npm run start -- --env local
```

Make sure to `npm run build` gasket-redux before copying!
